ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -781,14 +793,16 @@ static void HandleKernelRequest(void* messageMemory, uint32_t messageSize)
             }
             cout << endl;
 #endif
+            bool isDirectory = requestHeader->messageType == MessageType_KtoU_NotifyDirectoryRenamed;
             
-            if (relativePath != nullptr)
+            if (relativePath != nullptr || relativeFromPath != nullptr)
             {
                 result = HandleNewFileInRootNotification(
```

Comment:
We should only be calling `HandleNewFileInRootNotification` if `relativePath` is non-null, otherwise there's not a new file in the root.  If `relativePath` is null we should call `HandleFileNotification` instead to avoid the walk up the tree that checks for new directories.